feat: timeout

dormeiri-port opened the pull request:
1. Limit JQ running to 5 seconds by default
2. Add ability to pass the timeout
3. Small optimizations in recursive JQ running

dormeiri-port committed:
feat: timeout

ivankalinovski commented on lib/templateAsync.js:
```diff
@@ -64,15 +66,12 @@ const renderAsync =async (inputJson, template, execOptions = {}) => {
   const firstIndex = indices[0];
   if (indices.length === 1 && template.trim().startsWith('{{') && template.trim().endsWith('}}')) {
     // If entire string is a template, evaluate and return the result with the original type
-    return await jq.execAsync(inputJson, template.slice(firstIndex.start, firstIndex.end), execOptions);
```
why the await here is unnecessary? wont it confuse ts?

dormeiri-port replied:
await wrap the function with a promise, unless you need to catch the error, there is no reason to await, you can return the response directly.

ivankalinovski commented on src/binding.cc:
```diff
@@ -612,6 +614,23 @@ napi_value ExecAsync(napi_env env, napi_callback_info info) {
+        napi_throw_error(env, nullptr, "Invalid timeout_sec input");
+        return nullptr;
+      }
+      if(valuetype == napi_number){
+        status = napi_get_value_uint32(env, args[2], &work->timeout_sec);
+        if (status != napi_ok) {
+          napi_throw_error(env, nullptr, "Invalid timeout_sec input");
+          return nullptr;
+        }
+      }
```
should we prevent timeout 0?

dormeiri-port replied:
I think that it is fine to do timeout 0, wdyt? It just means that it will timeout immediately

ivankalinovski reviewed (commented):
LGTM

dormeiri-port committed:
chore(release): bump version to v1.0.2-rc.1

- Update package version from v1.0.1 to v1.0.2-rc.1 in package.json
- Update package-lock.json version to v1.0.2-rc.1
- Remove `nan` dependency from package-lock.json

ivankalinovski reviewed (approved):
LGTM
